### DISCUSSION TITLE
Update LibreChat with npm

### DISCUSSION THREAD

**Discussion by TCMAD** [danny-avila/LibreChat#1113] [Q&A] [answered]
Hello

I have difficulties updating LibreChat on v0.6.0 with the command `sudo npm run update`.

It returns me the following message :
```
WARNING: Error loading config file: /root/.docker/config.json: open /root/.docker/config.json: permission denied
ERROR: permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": dial unix /var/run/docker.sock: connect: permission denied
errors pretty printing info
Error: Docker is not running. You will need to start Docker Desktop or if using linux/mac, run `sudo systemctl start docker`
```

I precise that I'm able to use the v0.5.9 on my browser, the containers are on `UP` status and I followed the [Linux installation tutorial](https://github.com/danny-avila/LibreChat/blob/main/docs/install/linux_install.md#linux-installation-guide).

I noticed the client config files (`/root/.docker/config.json`) don't exist on my Linux, but I'm not sure if it's the main issue.

I'm not a docker expert, that's why I ask for help from the community. Do you have any idea on how to solve this?

**Comment by danny-avila** [comment, marked as answer]
> Hello
> 
> I have difficulties updating LibreChat on v0.6.0 with the command `sudo npm run update`.
> 
> It returns me the following message :
> 
> ```
> WARNING: Error loading config file: /root/.docker/config.json: open /root/.docker/config.json: permission denied
> ERROR: permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": dial unix /var/run/docker.sock: connect: permission denied
> errors pretty printing info
> Error: Docker is not running. You will need to start Docker Desktop or if using linux/mac, run `sudo systemctl start docker`
> ```
> 
> I precise that I'm able to use the v0.5.9 on my browser, the containers are on `UP` status and I followed the [Linux installation tutorial](https://github.com/danny-avila/LibreChat/blob/main/docs/install/linux_install.md#linux-installation-guide).
> 
> I noticed the client config files (`/root/.docker/config.json`) don't exist on my Linux, but I'm not sure if it's the main issue.
> 
> I'm not a docker expert, that's why I ask for help from the community. Do you have any idea on how to solve this?

Your logs say docker isn’t running. Check if it is, first. If the same command worked on 0.5.9 it should still work. Nothing was changed to the docker config

Since it’s a permissions error, you can try: `npm run update:sudo`

If you are using GitHub container images, Try using this guide instead: https://docs.librechat.ai/deployment/digitalocean.html

If you’re building the image yourself, you can try running the commands the script runs for you

```bash
git checkout main
git pull origin main
sudo docker rmi librechat:latest
# console.purple('Removing all unused dangling Docker images...');
sudo docker image prune -f
# console.purple('Building new LibreChat image...');
sudo docker-compose build
```

hope this helps

**Reply by TCMAD** [reply]
Thanks for your help :+1: .
Passing through git is easier than using npm in my case.
Perhaps you could add a link to this discussion to the `linux_install.md` in case other linux users have problems with future updates.

**Reply by danny-avila** [reply]
Will do! Glad it worked for you

**Reply by danny-avila** [reply]
I have added thorough instructions in the docs: https://github.com/danny-avila/LibreChat/pull/1117
